core/src/main/java/bio/terra/pearl/core/service/survey/SurveyParseUtils.java

```diff
@@ -131,6 +131,26 @@ public static SurveyQuestionDefinition unmarshalSurveyQuestion(
         return definition;
     }
 
+    private static Pattern surveyQuestionRegexPattern = Pattern.compile("\\{\\s*\\w+\\.[\\w.]+\\s*}");
+
+    public static Map<String, List<String>> findReferencedSurveyQuestions(Survey survey) {
+        return surveyQuestionRegexPattern.matcher(survey.getContent()).results().map(result -> {
+            String cleaned = result.group().substring(1, result.group().length() - 1).trim();
+
+            int periodIndex = cleaned.indexOf('.');
+
+            String surveyStableId = cleaned.substring(0, periodIndex);
+            String questionStableId = cleaned.substring(periodIndex + 1);
+
+            return Map.entry(surveyStableId, questionStableId);
+        }).filter(
+                entry -> !List.of("profile", "proxyProfile").contains(entry.getKey())
```

Comment:
pull this list out into a named constant -- and make sure there's a comment around line 315 of surveyUtil.js noting that any extraVariables that are objects need to be added here so they are not attempted to be fetched